mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,507 @@
+# Authors: Mark Wronkiewicz <wronk.mark@gmail.com>
+#          Jussi Nurminen <jnu@iki.fi>
+
+# This code was adapted and relicensed (with BSD form) with permission from
+# Jussi Nurminen
+
+# License: BSD (3-clause)
+
+# Equation numbers refer to Taulu and Kajola, 2005. 'Presentation of
+# electromagnetic multichannel data: The signal space separation method,' which
+# can be found for free online.
+
+# Note, there are an absurd number of different possible notations for
+# spherical coordinates, which confounds the notation for spherical harmonics.
+# Here, we purposefully stay away from shorthand notation in both and use
+# explicit terms (like 'azimuth' and 'polar') to avoid confusion.
+
+
+from __future__ import division
+from os import path as op
+import numpy as np
+from scipy.special import lpmv
+from scipy.linalg import pinv
+from scipy.misc import factorial as fact
+from ..utils import logger
+from .. import pick_types, pick_info
+from ..io.constants import FIFF
+from ..forward._compute_forward import _concatenate_coils
+from ..forward._make_forward import _read_coil_defs, _create_coils
+
+
+def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3):
+    """Apply Maxwell filter to data using spherical harmonics.
+
+    Parameters
+    ----------
+    raw : instance of mne.io.Raw
+        Data to be filtered
+    origin : array-like, shape (3,)
+        Origin of internal and external multipolar moment space in head coords
+        and in millimeters
+    int_order : int
+        Order of internal component of spherical expansion
+    ext_order : int
+        Order of external component of spherical expansion
+
+    Returns
+    -------
+    raw_sss : instance of mne.io.Raw
+        The raw data with Maxwell filtering applied
+    """
+
+    # TODO: Exclude 'bads' in multipolar moment calc, add back during
+    # reconstruction
+    assert (raw.info['bads'] == [], 'Maxwell filter does not yet handle bad '
+            'channels.')
+
+    # Read coil definitions from file
+    coil_def_file = op.join(op.split(__file__)[0], '..', 'data',
+                            'coil_def_Elekta.dat')
+    all_coils, meg_info = _make_coils(raw.info, accurate=True,
+                                      coil_def=coil_def_file)
```

Comment:
Our current coil def file is more or less complete. You are adding one with enhanced defs for a couple of coil types now. What you can do is modify the behavior of `_make_coils(...)` such that if you do `_make_coils(..., Elekta_defs=True)`, it will first search your enhanced file, and if it doesn't find the coils it needs, then it falls back to looking in the original/complete coil def file (and if not found there, error as before). Make sense?

In the code this could be done presumably by adding a `for` loop in the right place, where it loops over only one file (the current one; default) or the enhanced one and the current one (with a break if the proper coil is found) depending on the value of `Elekta_defs`.